tensorflow/core/graph/gpu_fusion_pass.cc

```diff
@@ -958,6 +1035,348 @@ void ROCmFusionOpBatchNormActivation::CreateFusionOp(const FusionData* d) {
 }
 //----------------------------------------------------------------------
 
+// ----------------------------------------------
+// ROCmFusionOpAddRelu implementation
+// ----------------------------------------------
+
+bool ROCmFusionOpAddRelu::DoFusion(const Node* n2) {
+  bool did_fusion = false;
+  FusionData d;
+  if (IsFusionEligible(n2, &d)) {
+    CreateFusionOp(&d);
+    did_fusion = true;
+  }
+  return did_fusion;
+}
+
+bool ROCmFusionOpAddRelu::IsFusionEligible(const Node* n2, FusionData* d) {
+  bool is_eligible = false;
+
+  // First check whether we have the right sequence of ops
+  if (isOpRelu(n2)) {  // "Relu" node
+    Node* n1 = nullptr;
+    TF_CHECK_OK(n2->input_node(0, &n1));
+    if (isOpAdd(n1)) {  // preceded by a "Add" op
+      d->add = n1;
+      d->relu = n2;
+
+      DumpNodeList(kVlogLevel,
+                   "Found Fusion Candidate Add+Relu: ", {d->add, d->relu});
+
+      is_eligible = true;
+    }
+  }
+
+  // ensure all the nodes are placed on the same GPU
+  if (is_eligible && !areAssignedToSameGpu({d->add, d->relu})) {
+    is_eligible = false;
+  }
+
+  // Next check if the first output of add node feeds a node other then
+  // the activation node
+  if (is_eligible) {
+    for (const Edge* e : d->add->out_edges()) {
+      if ((e->src_output() == 0) && (e->dst() != d->relu)) {
+        VLOG(kVlogLevel) << "\tSkipping Fusion : "
+                         << "Output from Add also feeds node other then Relu : "
+                         << e->dst()->name();
+        VLOG(kVlogLevel) << "===========";
+        is_eligible = false;
+        break;
+      }
+    }
+  }
+
+  // Next check if the datatype(s) are supported
+  if (is_eligible) {
+    is_eligible = false;
+
+    DataType T_add, T_relu;
+    TF_CHECK_OK(GetNodeAttr(d->add->def(), kAttr_T, &T_add));
+    TF_CHECK_OK(GetNodeAttr(d->relu->def(), kAttr_T, &T_relu));
+
+    // only float type is supported for now
```

Comment:
@deven-amd comment here needs to be revised?

Reply:
you are right....will fix in next commit